internal/runners/initialize/init.go

```diff
@@ -33,19 +32,49 @@ type RunParams struct {
 
 // Initialize stores scope-related dependencies.
 type Initialize struct {
-	out output.Outputer
+	config projectfile.ConfigGetter
+	out    output.Outputer
 }
 
 type primeable interface {
+	primer.Configurer
 	primer.Outputer
 }
 
 // New returns a prepared ptr to Initialize instance.
 func New(prime primeable) *Initialize {
-	return &Initialize{prime.Output()}
+	return &Initialize{prime.Config(), prime.Output()}
 }
 
-func sanitize(params *RunParams) error {
+// inferLanguage tries to infer a reasonable default language from the project currently in use
+// (i.e. `state use show`).
+// Error handling is not necessary because it's an input error to not include a language to
+// `state init`. We're just trying to infer one as a convenience to the user.
+func inferLanguage(config projectfile.ConfigGetter) string {
+	defaultProjectDir := config.GetString(constants.GlobalDefaultPrefname)
+	if defaultProjectDir == "" {
+		return ""
+	}
+	defaultProj, err := project.FromPath(defaultProjectDir)
+	if err != nil {
+		return ""
+	}
+	commitID := defaultProj.CommitUUID()
+	if commitID == "" {
+		return ""
+	}
+	lang, err := model.FetchLanguageForCommit(commitID)
+	if err != nil {
+		return ""
+	}
+	return fmt.Sprintf("%s@%s", lang.Name, lang.Version)
+}
+
+func sanitize(params *RunParams, config projectfile.ConfigGetter) error {
+	if params.Language == "" {
+		params.Language = inferLanguage(config)
```

Comment:
I recognize the code is already doing this, but ideally we should never override params, and consider them read-only. params should always be the thing the user passed in.

Suggest fixing this in the code for this runner while we're at it.

Reply:
The `sanitize` function is changing params for use in a later function. Any local values stored in this function will be discarded. Should I spend the time doing a more involved refactor?

Reply:
Yes please. Since this story is effectively about stabilising the init command I think it makes sense to make these architectural improvements.